ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/search_tree/algorithms/SearchTree.java

```diff
@@ -269,15 +277,23 @@ private void updateOptimalLeafWithNextDepthBestLeaf(AbstractNetworkPool networkP
                 }
                 try {
                     if (combinationFulfillingStopCriterion.isEmpty() || deterministicNetworkActionCombinationComparison(naCombination, combinationFulfillingStopCriterion.get()) < 0) {
-                        // Reset range action to their pre-perimeter set-points
-                        // TODO : instead of this parameter, clean rangeActions containted in Leaf. There should only be the optimized state's range actions
-                        // but then other RangeActionActivatinoResult in FARAO initialized with leaf's range actions should be initialized with prePerimeterOutput range Actions + leaf's
-                        // optimized range actions.
-                        input.getOptimizationPerimeter().getRangeActions().forEach(ra ->
-                            ra.apply(networkClone, input.getPrePerimeterResult().getRangeActionSetpointResult().getSetpoint(ra))
-                        );
-
-                        optimizeNextLeafAndUpdate(naCombination, networkClone);
+                        boolean shouldRangeActionBeRemoved = naCombinationsSorted.get(naCombination);
```

Comment:
Do you have a unit test that checks this new behaviour?

Reply:
No but we do have a new cucumber test in the branch `Better handling of max RA`.
This test has a `max-mip-iterations` value of 1.
With master branch, the test fails because it cannot converge fast enough (because it "forgot" the PST setpoints of the RootLeaf.
On our cucumber branch, however, it can find a better solution because it remembers the SPT setpoints of the rootLeaf.